### ISSUE TITLE
Linter: Implement `no-aria-label-misuse` rule

### ISSUE THREAD

**Issue by marcoroth** [marcoroth/herb#1223]
Implement the [`GitHub::Accessibility::NoAriaLabelMisuse`](https://github.com/github/erblint-github/blob/main/lib/erblint-github/linters/github/accessibility/no_aria_label_misuse.rb) rule from [`erblint-github`](https://github.com/github/erblint-github).

### Rule Details

This rule aims to discourage common misuse of the `aria-label` and `aria-labelledby` attributes. Support for these attributes is only guaranteed on interactive elements like `button` or `a`, or on static elements like `div` and `span` with a permitted `role`.

This rule will allow `aria-label` and `aria-labelledby` usage on `div` and `span` elements if set to a role other than the ones listed in [W3C: roles which cannot be named](https://w3c.github.io/aria/#namefromprohibited). This rule will never permit usage on `h1`–`h6`, `strong`, `i`, `p`, `b`, or `code`.

### Examples

**Incorrect** code for this rule:

```html
<span aria-label="This is a tooltip">I am some text.</span>
<div aria-labelledby="heading1">Goodbye</div>
<h1 aria-label="This will override the page title completely">Page title</h1>
```

**Correct** code for this rule:

```html
<button aria-label="Close">
  <svg src="closeIcon"></svg>
</button>
<span>Hello</span>
<div role="dialog" aria-labelledby="dialogHeading">
  <h1 id="dialogHeading">Heading</h1>
</div>
```

### References

- [erblint-github source](https://github.com/github/erblint-github/blob/main/lib/erblint-github/linters/github/accessibility/no_aria_label_misuse.rb)
- [erblint-github docs](https://github.com/github/erblint-github/blob/main/docs/rules/accessibility/no-aria-label-misuse.md)
- [Not so short note on aria-label usage](https://html5accessibility.com/stuff/2020/11/07/not-so-short-note-on-aria-label-usage-big-table-edition/)
- [Primer: Tooltip alternatives](https://primer.style/design/accessibility/tooltip-alternatives)

### Parent Issue

- #1205

**Comment by written-fresh** [comment]
@marcoroth I can take this

**Comment by marcoroth** [comment]
@written-fresh we might want to start prefixing these linter rules with `a11y-*`, see https://github.com/marcoroth/herb/issues/1310

**Comment by joelhawksley** [comment]
@written-fresh's PR seems abandoned, so I wrote https://github.com/marcoroth/herb/pull/1677

**Comment by written-fresh** [comment]
Yeah @joelhawksley I'm without a pc for the last bit. I haven't been able to finalize my draft. If yours works then great!
